import-service/src/handlers/importFileParser.ts

```diff
@@ -0,0 +1,71 @@
+import AWS from 'aws-sdk';
+import { S3Handler } from 'aws-lambda';
+import csv from 'csv-parser';
+import * as path from 'path';
+
+import config from '../../config';
+
+const importAndParseFileFromS3 = (
+    bucket: string,
+    objectKey: string,
+    s3: AWS.S3,
+): Promise<{[key: string]: string}[]> => new Promise((resolve, reject) => {
+    const params = {
+        Bucket: bucket,
+        Key: objectKey
+    };
+
+    const results = [];
+
+    s3.getObject(params)
+        .createReadStream()
+        .on('error', reject)
+        .pipe(csv())
+        .on('error', reject)
+        .on('data', (data) => results.push(data))
```

Comment:
I think storing chunks of streamed data in the array is not a good idea. Using streams helps us to process data bits by bits, so the memory footprint of the app would be tiny. But your implementation defeats its purpose by storing the whole file in memory. Imagine, for example, processing a CSV file over 3 gigabytes, this lambda function would be terminated due to memory overflow.

https://aws.amazon.com/about-aws/whats-new/2017/11/aws-lambda-doubles-maximum-memory-capacity-for-lambda-functions/#:~:text=AWS%20Lambda%20Doubles%20Maximum%20Memory%20Capacity%20for%20Lambda%20Functions,-Posted%20On%3A%20Nov&text=You%20can%20now%20allocate%203008MB,to%20your%20functions%20was%201536MB.